### ISSUE TITLE
I find <slot slot="..."> confusing

### ISSUE THREAD

**Issue by kenchris** [WICG/webcomponents#514]
What if we renamed `slot=` to `fill=`, so that it makes it clear that it fills the slot with that name. ie `<slot name=s2 fill=s1>` fill the slot s1 with another slot named s2

**Comment by hayatoito** [comment]
If someone has a strong concern about the current `slot=` attribute name, please +1.

We can not change the attribute name after shipping. Blink will ship Shadow DOM v1 after I resolve all v1 Shadow DOM spec issues.

**Comment by annevk** [comment]
I rather keep the current name. Otherwise we get fill vs assign, etc. Too much bikeshedding going on already.

**Comment by kenchris** [comment]
I don't think this is so much about bikeshedding about a name, but more about developer ergonomics. I found it pretty hard to follow the examples that @hayatoito made and had to spend quite some time understanding it. I am pretty sure that might be the case for many developers which is why I raised this issue.

**Comment by hayatoito** [comment]
I prefer the current `slot=` to `fill=`. `fill=` sounds a canvas-related attribute, reminding me of "fill a rectangle".

**Comment by kenchris** [comment]
I am not sure that `fill` is a good name either, but slot slot is just very confusing.

**Comment by annevk** [comment]
Can you elaborate on the confusion? It always seemed pretty clear and simple to me...

**Comment by rniwa** [comment]
`slot` attribute specifies to which `slot` element it belongs to.  It seems pretty clear to me.  The only alternative that maybe clearer might be `assignedslot` but that's such a long name for an attribute that may show up in a lot of places once you started using web components.

**Comment by treshugart** [comment]
As a consumer of the spec, I never found `<slot slot>` to be confusing in terms of naming. The confusing part for me was: what would happen when a slot is slotted into a slot? It seems instinctive that a `slot` attribute would correspond to a `<slot>` node. I'd have to have read documentation to find out what `fill` (or any other attribute for that matter) would mean.

**Comment by JanMiksovsky** [comment]
To add some data to this discussion: based on the experiences of the Basic Web Components project, we think it’s going to be quite rare to assign a slot directly to another slot.
- The vast majority of the components we write have only one slot, in which case no slot attribute will be specified. (The slot attribute is only required when targeting a specific slot in a component that has multiple slots.) We have approximately 50 or so components in our library in various stages of development, and of those, it looks like 3 will have multiple slots. We think the ability to support multiple slots is critical for certain situations, but in practice, at least, those are not typical. YMMV.
- Even in cases where you want to assign to a named slot, you’re often not assigning a slot element, but something else. That is, even when the slot attribute is used, it’s not applied directly to a slot element.
- In the rare cases when we’ve encountered situations where we will want to place a slot 1 in another component’s named slot 2, we will often want to wrap slot 1 in a containing element. Sometimes we want to add some additional elements to the ones defined by slot 1, or we may want to apply styling to the containing element. In cases like these, the slot attribute doesn’t end up being applied to a slot element:

```
<div id=“container" slot=“slot2”>
  … some elements …
  <slot name=“slot1”></slot>
</div>
```

Given all of the above, we think occurrences of <slot name=“slot1” slot=“slot2”> will be rare in practice.

**Comment by hayatoito** [comment]
It looks that no alternative name has been demonstrated to be clearly better, let me close this issue.
